core/textinput/src/textinput/Editor.cpp

```diff
@@ -243,29 +244,51 @@ namespace textinput {
     if (M < kCmd_END_TEXT_MODIFYING_CMDS) {
       PushUndo();
     }
-    if (fMode == kHistSearchMode) {
+    if (fMode == kHistRevSearchMode || fMode == kHistFwdSearchMode) {
+      const auto previousMode = fMode;
```

Comment:
We can merge the `if` statements in lines 256-260 and lines 276-280 here.
```suggestion
      const auto previousMode = fMode;
      fMode = (M == kCmdReverseSearch ? kHistRevSearchMode : kHistFwdSearchMode);
      if (previousMode != fMode) { // changed from fwd to bkw or viceversa, just update label
        SetHistSearchModePrompt(R.fDisplay);
        return kPRSuccess;
      }
```